### DISCUSSION TITLE
Change the IP for kafka bridge

### DISCUSSION THREAD

**Discussion by sreejesh123** [strimzi/strimzi-kafka-operator#5276] [Q&A]
hi, is it possible to use a specific IP for kafka bridge?   ours is a very specific use case where we have exposed only a set of ip which has the Firewall rule set. is it possible to set an ip range to kafka bridge?

**Comment by scholzj** [comment]
I guess the question is how do you expose it and what IP are you talking about. The Pod IP? The internal Kubernetes service IP? Some external IP?

Strimzi by default creates only the internal service for access within the same Kubernetes cluster. but any eternal addresses will be created by you directly - loadbalancer services, ingress, etc. So you should be able to set those your self if the infra supports it.

**Reply by sreejesh123** [reply]
thanks for a detailed answer.  Will the access be with in the namespace only or across the GKE cluster?    KAFKA/ZK are started on POD IP range  as you explained that is right behaviour and I was asking if we can give that a NODE IP range which in my case has firewall opened.  Though  after seeing your answer I will go with creating an Ingress or External LB for exposing the kafka to  on-prem apps.  thanks again

**Reply by scholzj** [reply]
I think you are mixing different things. You asked about Bridge first. Now you seem to talk about Kafka brokers. So I think you need to better explain what exactly are you trying to do.

**Reply by sreejesh123** [reply]
Apologies,  yes I did mix two things. My use case I want to start-up a KAFKA Bridge which can access with in Cluster (different namespaces on same Cluster) and Outside cluster but on same Subnet , and expose it to ON-PREM applicaiton interconnect too.    All the KAFKA/ZF are on POD ip range, the relevant service   mycluster-bootstrap,  my-kafka-ZK-client is on service ip range,   the bridge is starting on Service range IP.  I can think of only solution to my problem was to start the bridge on IP range which has Firewall rules relaxed.  which is my NOD/LB IP range.   My original Q was can I change the IP range of Bridge from Service IP  to NODE IP range?    If I am correct BRIDGE is type: cluster IP can I change this to LB type at all?  any example yaml you can share if its possible?

**Reply by scholzj** [reply]
I do not have an example. But you can just manually create your own service called for example `<my-bridge>-loadbalancer` or `<my-bridge>-nodeport` and set there any service type you want (and possibly set there the load balancer IP etc.). Since it will be a custom name, the operator should not care about it and you can use it with your own settings.

**Reply by sreejesh123** [reply]
thank you, to confirm on one point. If I have two namespace with in same Cluster ,  does by default kafka bridge is accessible on both namespace?  or will it again depend on how my Subnet and FW are set?  Sorry new in cloud too. hence the Q

**Reply by scholzj** [reply]
A lot of this depends on your cluster and its configuration. The default and probably most common setup is that your whole cluster is one virtual network and you can communicate between namespaces easily. You can also create NetworkPolicies to restrict who can access your application. But there are many different networking plugins which sometimes do things differently. So it might not apply in all cases. If you created your Kube cluster your self, the above should most probably apply. If someone else manages it for you, it might be best to check with the Kube cluster admins.

**Reply by sreejesh123** [reply]
Thanks for all the support so far Jakub.   I have my kafka-bridge up and running and created a LB to access the bridge from my local app box.   I am trying the commands detailed on https://strimzi.io/blog/2019/11/05/exposing-http-bridge/    I am seeing all the responses correct as documented on the page except the consumer one. 

curl -X POST XXXXXXX/consumers/my-group/instances/my-consumer/subscription   -H 'content-type: application/vnd.kafka.v2+json'   -d '{
    "topics": [
        "my-topic"
    ]
}' -v


I am getting below message :

> POST /consumers/my-group/instances/my-consumer/subscription HTTP/1.1
> Host: 100.88.4.119:443
> User-Agent: curl/7.64.0
> Accept: */*
> content-type: application/vnd.kafka.v2+json
> Content-Length: 44
>
} [44 bytes data]
* upload completely sent off: 44 out of 44 bytes
< HTTP/1.1 204 No Content
<
100    44    0     0  100    44      0    140 --:--:-- --:--:-- --:--:--   140



My Q -

which log file should I check to see what really happened and why this is replying 204 No Content (when I am sure the content does exist)..   To help me with debug.  

Is there any cheat sheet which gives some commands and logs which we need to monitor for different error type?

Sorry as said earliest very new to Cloud and Strimzi hence this request.

**Reply by scholzj** [reply]
I'm not sure the POST for subscribing returns something. So HTTP 204 might be correct here. The docs suggest it is correct: https://strimzi.io/docs/bridge/latest/#_subscribe (the blog post might be outdated - it is from 2019). To get the current subscriptions, you should use the GET call: https://strimzi.io/docs/bridge/latest/#_listsubscriptions

The logs for the bridge would be in the Bridge pod logs - you can use `kubectl logs`, Kubernetes Dashboard or whatever log collection system you have in your cluster to see them.
